packages/wonder-blocks-clickable/components/clickable.js

```diff
@@ -0,0 +1,174 @@
+
+type Props = {|
+    /**
+     * The child of Clickable must be a funciton which returns the component
+     * which should be made Clickable.
+     */
+    children: (eventState: ClickableState) => React.Node,
+
+    /**
+     * An onClick function which Clickable can execure
+     */
+    onClick?: (e: SyntheticEvent<>) => mixed,
+
+    /**
+     * Optinal href which Clickable should direct to, uses Client-side routing
+     * by default if react-router is present
+     */
+    href?: string,
+
+    /**
+     * Styles to apply to the Clickable compoenent
+     */
+    style?: StyleType,
+
+    /**
+     * Disables or enables the child, defaults to false
+     */
+    disabled: boolean,
+
+    /**
+     * The role of the component
+     */
+    role?: ClickableRole,
+
+    /**
+     * If href is present skipClientNav will avoid client-side routing
+     */
+    skipClientNav?: boolean,
+
+    "aria-label": string,
+    testId?: string,
+|};
+
+const StyledAnchor = addStyle<"a">("a");
+const StyledButton = addStyle<"button">("button");
+const StyledLink = addStyle<typeof Link>(Link);
+
+export default class Clickable extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+        "aria-label": "",
+    };
+
+    static contextTypes = {router: PropTypes.any};
+
+    render() {
+        const {
+            "aria-label": ariaLabel, // eslint-disable-line react/prop-types
+            children,
+            disabled,
+            href,
+            onClick,
+            role,
+            skipClientNav,
+            style,
+            testId,
+        } = this.props;
+        const {router} = this.context;
+
+        const ClickableBehavior = getClickableBehavior(
+            href,
+            skipClientNav,
+            router,
+        );
+
+        return (
+            <ClickableBehavior>
+                {(state, handlers) => {
+                    const commonProps = {
+                        "aria-label": ariaLabel,
+                        "data-test-id": testId,
+                        role: role,
+                        style: [styles.reset, style],
+                        ...handlers,
+                    };
+
+                    const content = children(state);
+                    const activeHref = href && !disabled;
+                    const useClientRouter = router && !skipClientNav;
+
+                    if (activeHref && useClientRouter) {
+                        return (
+                            <StyledLink
+                                {...commonProps}
+                                to={href}
+                                onClick={onClick}
+                                aria-disabled={disabled ? "true" : undefined}
+                            >
+                                {content}
+                            </StyledLink>
+                        );
+                    } else if (activeHref && !useClientRouter) {
```

Comment:
This can just be an `if` since the previous condition returns.

Reply:
I feel like it’s better this way for clarity, what do you think?